### PR TITLE
#161966868 Setup Sass Preprocessor

### PR DESCRIPTION
**Pull request by leonardnjura**
#### What does this PR do?
- Integrates `.sass` and `.scss` style compiler
- Allows app to import `.scss`
- Allows app to build or compile into `.css` 

#### How should this be manually tested?
`$ git clone -b ch-sass-setup-161966868 https://github.com/andela/ah-infinity-stones-frontend`
`$ cd ah-infinity-stones-frontend`
> Install node modules specified in `package.json`
`$ npm install`
`$ npm start`
> Now you are ready to edit and see own changes
> Navigate to `./src/css/` directory and edit the `main.scss`
> Import this stylesheet in `index.js` below all styles with import './css/main.scss';` 
> Now you should see the page relaod with new style
> `$ npm run build-css` to build css
> Now you should see the compiled `.css` with the name used above

#### What are the relevant pivotal tracker stories?
[#161966868](https://www.pivotaltracker.com/story/show/161966868)

#### Screenshots
screen shot 2019-01-11 at 12 11 23

screen shot 2019-01-11 at 12 14 16

screen shot 2019-01-09 at 13 28 30

screen shot 2019-01-11 at 10 10 21

### CODE REVIEW

**Comment by kamyaD** [comment]
@leonardnjura I think the task is well done. Thanks for your effort.

**Review by PromasterGuru — APPROVED** [review]

**Review by PromasterGuru — COMMENTED** [review]
I think the first src/App.css was awesome compared to the current one. Change in the formatting wasn't necessary. However, that doesn't affect the functionality. Good work though.

**Review by rkemmy — APPROVED** [review]
Great job with the setup. It's also a nice touch shaking things up with the colors. It's a good way to get things started on the design.

**Review by codeplus254 — APPROVED** [review]
Great work with the sass. Now we don't have to change all .css files when updating the look of the site

File: package.json [hunk 1]
```diff
@@ -3,15 +3,19 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "file-loader": "^3.0.1",
```
Comment (johnmutuma5) [inline package.json]: @leonardnjura, shouldn't this be a devDependency?

File: src/index.js [hunk 2]
```diff
@@ -3,6 +3,9 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
+import './css/font-awesome.min.css'
+import './css/materialdesignicons.min.css';
+import './css/style_1.css';
```
Comment (johnmutuma5) [inline src/index.js]: @leonardnjura, we want to be able to import `.scss` or `.sass` files. I can only see `.css` files used in the project. 🤔

**Review by johnmutuma5 — CHANGES_REQUESTED** [review]
@leonardnjura, good progress. Please see my reviews below.

**Comment by ThaDeveloper** [comment]
@leonardnjura Good progress, it would be more appropriate to package both the css and fonts folders into a 'static' folder. Also, check on the merge conflict

File: src/index.js (hunk 2 above)
Comment (leonardnjura) [reply src/index.js]: @johnmutuma5 thanks for the observation. The first two `.css` files highlighted are for UI icons and we cannot compile these with sass. The last `.css` file highlighted is the compiled version. The directory `~src/css` has the `.sass` or `.scss` creating the file imported on line `8`

Comment (leonardnjura) [reply src/index.js]: @johnmutuma5 every `$ npm start` listens to the css directory for changes in the `.sass` or `.scss` and updates/compiles into the respective `.css`

File: package.json (hunk 1 above)
Comment (leonardnjura) [reply package.json]: @johnmutuma5 thanks for raising this. I kinda used `$ yarn add` without a flag, I will fix right away and alert you

File: src/index.js (hunk 2 above)
Comment (johnmutuma5) [reply src/index.js]: @leonardnjura, does that mean that we will be keeping both types, a `.css` and a `.sass` file all containing similar rules but different syntax? 🤔

**Comment by PromasterGuru** [comment]
Hey @leonardnjura, check your indentation and also terminate your line 6 in index.js imports with a semicolon to solve the error on reflected imports. Also, work on the merge conflicts and other violations.

File: src/index.js (hunk 2 above)
Comment (leonardnjura) [reply src/index.js]: Yes. Perhaps to add light @johnmutuma5  i added a hot reload on scss edit with `npm run build-css` so if we go by the fact that `.sass` or `.scss`allows us to code minimal and it auto compiles into a pretty css

Comment (leonardnjura) [reply src/index.js]: These scripts are inside of bin directory after i refactore some clean up.

Comment (johnmutuma5) [reply src/index.js]: @leonardnjura, I don't think that we really should keep different types of the same `css` styling rules. Is there a way that the current set up can only have `.sass` or `.scss` and not include a `.css` version of the same? Ideally, we should only have one.

**Review by kamyaD — COMMENTED** [review]
@leonardnjura Great work, However hound seems to have a problem with import and class. Please have a look.

**Review by johnmutuma5 — APPROVED** [review]

**Pull request merged**